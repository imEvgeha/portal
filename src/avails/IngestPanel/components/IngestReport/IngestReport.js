import React from 'react';
import PropTypes from 'prop-types';
import Constants from '../../Constants';
import './IngestReport.scss';

const IngestReport = ({report}) => {
    const reportFields = Constants.REPORT;
    const reportValues = report || {};
    return (
        <div className='ingest-report'>
            {
                Object.keys(reportFields).map(key => (
                    <div className='ingest-report__field'>
                        <span className='ingest-report__field--label'>{reportFields[key]}</span>
                        <span className='ingest-report__field--value'>{reportValues[key] || 0}</span>
                    </div>
                ))
            }
        </div>
    );
};

IngestReport.propTypes = {
    report: PropTypes.object
};

IngestReport.defaultProps = {
    report: {}
};

export default IngestReport;