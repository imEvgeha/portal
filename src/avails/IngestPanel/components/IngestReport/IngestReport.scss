.ingest-report {
  display: inline-flex;
  &__field {
    display: flex;
    flex-direction: column;
    align-items: center;
    color: #42526E;
    font-size: 11px;
    width: 45px;
    height: 45px;
    &--label {
      margin-bottom: 5px;
    }
    &--value {
      background: #A5ADBA;
      border-radius: 32px;
      padding: 2px 8px;
      text-align: center;
      font-weight: bold;
    }
  }
}