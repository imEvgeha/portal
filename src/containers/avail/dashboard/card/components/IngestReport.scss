.nexus-c-ingest-report{
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  font-size: 12px;
  .nexus-c-ingest-report__counters{
    display: flex;
    margin-top: 5px;
    .nexus-c-ingest-report__counters-section {
      margin-right: 10px;
      flex-shrink: 0;
      display: flex;
      flex-direction: column;
      align-items: center;
      label {
        font-size: 16px;
        margin-bottom: 0;
      }
      .nexus-c-ingest-report__counters-section--value {
        font-size: 20px;
        font-weight: bold;
      }
    }
  }
}
.Tooltip{
  max-width: none;
}