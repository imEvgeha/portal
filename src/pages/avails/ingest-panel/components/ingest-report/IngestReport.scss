@import '../../../../../styles/atlaskit/reset';
@import '../../../../../styles/atlaskit/theme';
@import '../../../../../styles/atlaskit/mixins';

.ingest-report {
    width: 100%;
    @include addSpacing(margin-top, 3.5);
    display: inline-flex;
    flex-direction: column;

    &__fields {
        display: inline-flex;
        justify-content: space-between;
    }

    &__field {
        display: flex;
        flex-direction: column;
        align-items: center;
        color: map-get($dark-neutral-colors, dark-neutral-5);
        @include addSpacing(width, 10);
        @include addSpacing(height, 10);
        line-height: initial;

        &--label {
            @include heading(h100);
            @include margin(0, 0, 1, 0);
        }

        &--value {
            display: flex;
            align-items: center;
            justify-content: center;
            color: map-get($dark-neutral-colors, dark-neutral-1);
            background: map-get($light-neutral-colors, light-neutral-1);
            border-radius: 32px;
            padding: 2px 8px;
            text-align: center;
            font-weight: bold;
            min-width: 22px;
            min-height: 20px;
        }
    }

    &__error-message {
        font-size: 11px;
        color: #de350b;
        margin: 10px 0 0 5px;
    }
}
