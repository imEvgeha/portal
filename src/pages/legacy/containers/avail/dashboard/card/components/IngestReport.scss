.nexus-c-ingest-report {
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    font-size: 12px;
    &__counters {
        display: flex;
        margin-top: 5px;
    }
    &__counters-section {
        margin-right: 10px;
        flex-shrink: 0;
        display: flex;
        flex-direction: column;
        align-items: center;
        label {
            font-size: 16px;
            margin-bottom: 0;
        }
    }
    &__counters-section--value {
        font-size: 20px;
        font-weight: bold;
    }
}
.Tooltip {
    max-width: none;
}
