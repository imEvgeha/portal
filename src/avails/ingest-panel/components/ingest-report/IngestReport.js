import React, {useState} from 'react';
import PropTypes from 'prop-types';
import Constants from '../../constants';
import './IngestReport.scss';

const IngestReport = ({report, showErrorMessage = true, filterClick}) => {
    const [activeFilter, setActiveFilter] = useState('');
    const reportFields = Constants.REPORT;
    const reportValues = report || {};
    const onFilterClick = filterKey => {
        const key = activeFilter === filterKey ? '' : filterKey;
        filterClick(reportFields[key] !== Constants.REPORT.total && reportFields[key]);
        setActiveFilter(key);
    };

    const FILTERABLE_KEYS = ['total', 'pending', 'errors'];

    return (
        <div className='ingest-report'>
            <div className='ingest-report__fields'>
                {
                    Object.keys(reportFields).map(key => (
                        <div className='ingest-report__field' key={key}>
                            <span className='ingest-report__field--label'>{reportFields[key]}</span>
                            <span
                                className={`ingest-report__field--value ${(activeFilter === key) ? 'filter-active' : ''}`}
                                onClick={() => FILTERABLE_KEYS.includes(key) && onFilterClick(key)} >
                                {reportValues[key] || ''}
                            </span>
                        </div>
                    ))
                }
            </div>
            {
                showErrorMessage && report.errorDetails && (
                    <span className='ingest-report__error-message'>
                        {report.errorDetails}
                    </span>
                )
            }
        </div>
    );
};

IngestReport.propTypes = {
    report: PropTypes.object,
    showErrorMessage: PropTypes.bool,
    filterClick: PropTypes.func,
};

IngestReport.defaultProps = {
    report: {},
    showErrorMessage: true,
    filterClick: () => null,
};

export default IngestReport;
