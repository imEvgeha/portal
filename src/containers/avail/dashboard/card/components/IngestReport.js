import React from 'react';
import NexusTooltip from '../../../../../ui-elements/nexus-tooltip/NexusTooltip';
import Constants from '../Constants';
import './IngestReport.scss';

function IngestReport(params) {
    const {valueFormatted, data: {ingestReport}} = params;
    const content = (
        <div className='nexus-c-ingest-report'>
            <div>{valueFormatted}</div>
            <div className='nexus-c-ingest-report__counters'>
                {
                    Constants.REPORT_FIELDS.map(({field, displayName}) => (
                        <div key={field} className='nexus-c-ingest-report__counters-section'>
                            <label>{displayName}</label>
                            <div className='nexus-c-ingest-report__counters-section--value'>
                                {ingestReport[field]}
                            </div>
                        </div>
                    ))
                }
            </div>
        </div>
    );
    return (
        <NexusTooltip content={content}>
            <div>{valueFormatted}</div>
        </NexusTooltip>
    );
}

export default IngestReport;
