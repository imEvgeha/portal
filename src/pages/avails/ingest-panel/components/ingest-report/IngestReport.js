import React, {useState, useEffect} from 'react';
import PropTypes from 'prop-types';
import Constants from '../../constants';
import './IngestReport.scss';
import RightsURL from '../../../../legacy/containers/avail/util/RightsURL';

const IngestReport = ({report, showErrorMessage = true, filterClick, ingestId}) => {
    const [activeFilter, setActiveFilter] = useState('total');
    const reportFields = Constants.REPORT;
    const reportValues = report || {};
    const onFilterClick = filterKey => {
        const key = activeFilter === filterKey ? 'total' : filterKey;
        filterClick(reportFields[key].value);
        setActiveFilter(key);
    };

    useEffect(() => {
        onFilterClick('total');
    }, [report]);

    const FILTERABLE_KEYS = ['total', 'pending', 'errors'];

    return (
        <div className="ingest-report">
            <div className="ingest-report__fields">
                {
                    Object.keys(reportFields).map(key => (
                        <div className="ingest-report__field" key={key}>
                            <span className="ingest-report__field--label">{reportFields[key].label}</span>
                            <span
                                className={`ingest-report__field--value ${(activeFilter === key) ? 'filter-active' : ''}`}
                                onClick={() => (key === 'fatal' ? window.open(RightsURL.getFatalsRightsSearchUrl(ingestId), '_blank') : FILTERABLE_KEYS.includes(key) && onFilterClick(key))}
                            >
                                {reportValues[key] || 0}
                            </span>
                        </div>
                    ))
                }
            </div>
            {
                showErrorMessage && report.errorDetails && (
                    <span className="ingest-report__error-message">
                        {report.errorDetails}
                    </span>
                )
            }
        </div>
    );
};

IngestReport.propTypes = {
    report: PropTypes.object,
    showErrorMessage: PropTypes.bool,
    filterClick: PropTypes.func,
    ingestId: PropTypes.string,
};

IngestReport.defaultProps = {
    report: {},
    showErrorMessage: true,
    filterClick: () => null,
    ingestId: '',
};

export default IngestReport;
