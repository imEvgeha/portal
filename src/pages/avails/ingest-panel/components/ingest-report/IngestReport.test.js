/* eslint-disable no-magic-numbers */
import React from 'react';
import {shallow} from 'enzyme';
import NexusTooltip from '../../../../../ui/elements/nexus-tooltip/NexusTooltip';
import IngestReport from './IngestReport';

describe('IngestReport', () => {
    let wrapper = null;
    const errorText = 'Some attachments could not be processed';
    let props = {};

    beforeEach(() => {
        props = {
            filterClick: () => null,
            report: {
                total: 20001,
                success: 1000,
                created: 1001,
                updated: 19999,
                fatal: 3,
                errors: 2,
                pending: 0,
                errorDetails: errorText,
            },
            ingestId: 'avih_zLTMLyei6L',
        };
        wrapper = shallow(<IngestReport {...props} />);
    });

    it('should match snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should include an ingest-report and ingest-report__fields  div', () => {
        const ingestReport = wrapper.find('.ingest-report');
        expect(ingestReport).toHaveLength(1);
        const ingestReportFields = wrapper.find('.ingest-report__fields');
        expect(ingestReportFields).toHaveLength(1);
    });

    it('should include error message', () => {
        const errorMessage = wrapper.find('.ingest-report__error-message');
        expect(errorMessage).toHaveLength(1);
        expect(errorMessage.text()).toEqual(errorText);
    });

    it('should include error message', () => {
        const labels = wrapper.find('.ingest-report__field--label');
        expect(labels).toHaveLength(6);
        const values = wrapper.find('.ingest-report__field--value');
        expect(values).toHaveLength(6);
        expect(labels.at(0).text()).toEqual('Rights');
        expect(values.at(0).text()).toEqual('20001');

        expect(labels.at(1).text()).toEqual('New');
        expect(values.at(1).text()).toEqual('1001');

        expect(labels.at(2).text()).toEqual('Updated');
        expect(values.at(2).text()).toEqual('19999');

        expect(labels.at(3).text()).toEqual('Fatals');
        expect(values.at(3).text()).toEqual('3');

        expect(labels.at(4).text()).toEqual('Pending');
        expect(values.at(4).text()).toEqual('0');

        expect(labels.at(5).text()).toEqual('Errors');
        expect(values.at(5).text()).toEqual('2');
    });

    it('should not include any tooltip', () => {
        const tooltips = wrapper.find(NexusTooltip);
        expect(tooltips).toHaveLength(0);
    });

    it('should include tooltip', () => {
        const updatedProps = {...props, hasTooltips: true};
        wrapper = shallow(<IngestReport {...updatedProps} />);
        const tooltips = wrapper.find(NexusTooltip);
        expect(tooltips).toHaveLength(2);
    });

});

