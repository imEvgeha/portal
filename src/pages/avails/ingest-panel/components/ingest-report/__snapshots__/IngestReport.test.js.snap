// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IngestReport should match snapshot 1`] = `
<div
  className="ingest-report"
>
  <div
    className="ingest-report__fields"
  >
    <div
      className="ingest-report__field"
      key="total"
    >
      <span
        className="ingest-report__field--label"
      >
        Rights
      </span>
      <span
        className="ingest-report__field--value filter-active"
        onClick={[Function]}
      >
        20001
      </span>
    </div>
    <div
      className="ingest-report__field"
      key="created"
    >
      <span
        className="ingest-report__field--label"
      >
        New
      </span>
      <span
        className="ingest-report__field--value"
        onClick={[Function]}
      >
        1001
      </span>
    </div>
    <div
      className="ingest-report__field"
      key="updated"
    >
      <span
        className="ingest-report__field--label"
      >
        Updated
      </span>
      <span
        className="ingest-report__field--value"
        onClick={[Function]}
      >
        19999
      </span>
    </div>
    <div
      className="ingest-report__field"
      key="fatal"
    >
      <span
        className="ingest-report__field--label"
      >
        Fatals
      </span>
      <span
        className="ingest-report__field--value"
        onClick={[Function]}
      >
        3
      </span>
    </div>
    <div
      className="ingest-report__field"
      key="pending"
    >
      <span
        className="ingest-report__field--label"
      >
        Pending
      </span>
      <span
        className="ingest-report__field--value"
        onClick={[Function]}
      >
        0
      </span>
    </div>
    <div
      className="ingest-report__field"
      key="errors"
    >
      <span
        className="ingest-report__field--label"
      >
        Errors
      </span>
      <span
        className="ingest-report__field--value"
        onClick={[Function]}
      >
        2
      </span>
    </div>
  </div>
  <span
    className="ingest-report__error-message"
  >
    Some attachments could not be processed
  </span>
</div>
`;
